plugin.py

```diff
@@ -25,13 +29,21 @@ class LspTypescriptPlugin(NpmClientHandler):
     def install_in_cache(cls) -> bool:
         return False
 
-    def on_ready(self, api) -> None:
-        api.on_request('_typescript.rename', self._on_typescript_rename)
-
-    def _on_typescript_rename(self, textDocumentPositionParams, respond):
-        view = sublime.active_window().open_file(
-            uri_to_filename(textDocumentPositionParams['textDocument']['uri'])
-        )
+    @classmethod
+    def is_allowed_to_start(
+        cls,
+        window: sublime.Window,
+        initiating_view: Optional[sublime.View] = None,
+        workspace_folders: Optional[List[WorkspaceFolder]] = None,
+        configuration: Optional[ClientConfig] = None
+    ) -> Optional[str]:
+        if not workspace_folders:
```

Comment:
What if we try to find the root of the project and make the LS happy(instead of quitting where there is not one)?

```python
def find_project_root(self, view):
    if view is None or view.file_name() is None:
        return ""

    # Starting at the path of the current file, keep going up one folder
    # until we land in a folder with a cabal file in it, or we hit the
    # root.
    cwd = os.path.dirname(view.file_name())
    while cwd != "/" and not glob.glob(os.path.join(cwd, "*.package.json")):  # or jsconfig.json, or tsconfig.json
        cwd = os.path.abspath(cwd + "/../")

    # We didn't find a cabal file, so return the empty string.
    if cwd == "/":
        return ""

    # Return the found location
    return cwd
```
this was inspired by some code OdatNurd left on the discord channel.

Reply:
We can't change workspace root on ST3 though. So that would only potentially work on ST4.